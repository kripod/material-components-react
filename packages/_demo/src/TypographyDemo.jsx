import React from 'react';
import Typography from '../../typography';

const TypographyDemo = () =>
  <div>
    <h1>Typography</h1>

    {[false, true].map(adjustMargin =>
      <div key={adjustMargin}>
        <h2>{adjustMargin ? 'With' : 'Without'} margin adjustments</h2>

        <section>
          <Typography adjustMargin={adjustMargin}>
            <h1><Typography display4>Display 4</Typography></h1>
            <h1><Typography display3>Display 3</Typography></h1>
            <h1><Typography display2>Display 2</Typography></h1>
            <h1><Typography display1>Display 1</Typography></h1>
            <h1><Typography headling>Headline</Typography></h1>

            <h2>
              <Typography title>
                Title <Typography caption>Caption.</Typography>
              </Typography>
            </h2>

            <h2><Typography subheading2>Subheading 2</Typography></h2>
            <h2><Typography subheading1>Subheading 1</Typography></h2>

            <p>
              <Typography body1>
                Body 1 paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
                quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
                fugiat nulla pariatur.
              </Typography>
            </p>

            <aside><Typography body2>Body 2 text, calling something out.</Typography></aside>
          </Typography>
        </section>
      </div>,
    )}
  </div>
;

export default TypographyDemo;
