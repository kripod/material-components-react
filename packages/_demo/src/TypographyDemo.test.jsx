import React from 'react';
import renderer from 'react-test-renderer';
import TypographyDemo from './TypographyDemo';

it('renders correctly', () => {
  const tree = renderer.create(<TypographyDemo />).toJSON();
  expect(tree).toMatchSnapshot();
});
