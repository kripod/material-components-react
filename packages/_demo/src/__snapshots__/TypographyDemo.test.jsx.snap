exports[`test renders correctly 1`] = `
<div>
  <h1>
    Typography
  </h1>
  <div>
    <h2>
      Without
       margin adjustments
    </h2>
    <section>
      <span
        className="mdc-typography ">
        <h1>
          <span
            className="mdc-typography mdc-typography--display4 ">
            Display 4
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display3 ">
            Display 3
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display2 ">
            Display 2
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display1 ">
            Display 1
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography "
            headling={true}>
            Headline
          </span>
        </h1>
        <h2>
          <span
            className="mdc-typography mdc-typography--title ">
            Title 
            <span
              className="mdc-typography mdc-typography--caption ">
              Caption.
            </span>
          </span>
        </h2>
        <h2>
          <span
            className="mdc-typography "
            subheading2={true}>
            Subheading 2
          </span>
        </h2>
        <h2>
          <span
            className="mdc-typography "
            subheading1={true}>
            Subheading 1
          </span>
        </h2>
        <p>
          <span
            className="mdc-typography mdc-typography--body1 ">
            Body 1 paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
          </span>
        </p>
        <aside>
          <span
            className="mdc-typography mdc-typography--body2 ">
            Body 2 text, calling something out.
          </span>
        </aside>
      </span>
    </section>
  </div>
  <div>
    <h2>
      With
       margin adjustments
    </h2>
    <section>
      <span
        className="mdc-typography mdc-typography--adjust-margin ">
        <h1>
          <span
            className="mdc-typography mdc-typography--display4 ">
            Display 4
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display3 ">
            Display 3
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display2 ">
            Display 2
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography mdc-typography--display1 ">
            Display 1
          </span>
        </h1>
        <h1>
          <span
            className="mdc-typography "
            headling={true}>
            Headline
          </span>
        </h1>
        <h2>
          <span
            className="mdc-typography mdc-typography--title ">
            Title 
            <span
              className="mdc-typography mdc-typography--caption ">
              Caption.
            </span>
          </span>
        </h2>
        <h2>
          <span
            className="mdc-typography "
            subheading2={true}>
            Subheading 2
          </span>
        </h2>
        <h2>
          <span
            className="mdc-typography "
            subheading1={true}>
            Subheading 1
          </span>
        </h2>
        <p>
          <span
            className="mdc-typography mdc-typography--body1 ">
            Body 1 paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
          </span>
        </p>
        <aside>
          <span
            className="mdc-typography mdc-typography--body2 ">
            Body 2 text, calling something out.
          </span>
        </aside>
      </span>
    </section>
  </div>
</div>
`;
